.neuro/live-s3.yml

```
@@ -0,0 +1,314 @@
+    dockerfile: $[[ flow.workspace ]]/label_studio/label_studio.Dockerfile
+    context: $[[ flow.workspace ]]/label_studio
+    build_preset: cpu-medium
+
+jobs:
+  prepare_remote_dataset:
+    name: $[[ flow.title ]]-prepare-dataset
+    image: neuromation/neuro-extras:20.12.15
+    detach: False
+    volumes:
+      - ${{ volumes.remote_dataset.ref_rw }}
+    params:
+      dataset_path:
+        default: http://vision.stanford.edu/aditya86/ImageNetDogs/images.tar
+        descr: URL of dataset to download
+    bash: |
+      DST=$(basename ${{ params.dataset_path }} .tar)
+      if [ ! -e ${{ volumes.remote_dataset.mount }}/$DST ]; then
+        neuro-extras data cp --extract --use-temp-dir ${{ params.dataset_path }} ${{ volumes.remote_dataset.mount }}/$DST;
+      else
+        echo "Dataset is already downloaded, skipping."
+      fi
+
+  extend_data:
+    name: $[[ flow.title ]]-extend-data
+    image: $[[ images.train.ref ]]
+    detach: False
+    pass_config: True
+    volumes:
+      - ${{ volumes.remote_dataset.ref_ro }}
+      - secret:gh-rsa:/root/.ssh/id-rsa
+    env:
+      PROJECT: /usr/project
+      PYTHONPATH: /usr/project
+    params:
+      extend_dataset_by:
+        default: "1"
+        descr: |
+          How many new images to add into current dataset.
+    bash: |
+      git clone -b ${PROJECT_GIT_BRANCH} ${PROJECT_GIT_REPO} ${PROJECT} > /dev/null 2>&1
+
+      export DATA_PATH=${PROJECT}/data/images
+
+      # Init S3
+      mkdir -p $DATA_PATH
+      [[ -z $(neuro blob ls blob:${BUCKET_NAME}) ]] && echo "Bucket is empty" \
+        || neuro blob cp -rT blob:${BUCKET_NAME} $(dirname ${DATA_PATH}) # download data if the bucket is not empty
+
+      echo "Extending dataset"
+      python ${PROJECT}/label_studio/extend_dataset.py \
+        --cur_dataset ${DATA_PATH} \
+        --full_dataset ${{ volumes.remote_dataset.mount }}/images/Images/ \
+        --nmber_of_imgs ${{ params.extend_dataset_by }}
+
+      # Upload dataset to S3"
+      neuro blob cp -ru ${DATA_PATH} blob:${BUCKET_NAME}
+
+      # Validate dataset:
+      neuro blob ls -r blob:${BUCKET_NAME}
+      echo "Total images: $(neuro blob ls -r blob:${BUCKET_NAME} | grep -oP '\w*\.\w+$' | wc -l)" # grep files only
+
+  label_studio:
+    image: $[[ images.label_studio.ref ]]
+    name: $[[ flow.title ]]-label-studio
+    http_port: 443
+    http_auth: False
+    life_span: 1d
+    pass_config: True
+    detach: False
+    browse: True
+    volumes:
+      - secret:gh-rsa:/root/.ssh/id-rsa
+    env:
+      PROJECT: /usr/project
+      DATA_PATH: /usr/project/data/Images
+      LABEL_STUDIO_LOCAL_FILES_SERVING_ENABLED: True
+      LABEL_STUDIO_PROJECT: /label-studio-project
+      LS_TOKEN: secret:ls-token
+      LS_PORT: 443
+    bash: |
+      git clone -b ${PROJECT_GIT_BRANCH} ${PROJECT_GIT_REPO} ${PROJECT}
+
+      # Pull from S3
+      neuro blob cp blob:${BUCKET_NAME}/label_studio.sqlite3 "${PROJECT}/data/label_studio.sqlite3"
```

```suggestion
      neuro blob cp blob:${BUCKET_NAME}/label_studio.sqlite3 "${PROJECT}/data/label_studio.sqlite3" || echo Label Studio data was not found
```
in case if someone launches LS for the first time in this project